docs/user/production-considerations/task-manager-troubleshooting.asciidoc

```diff
@@ -0,0 +1,710 @@
+
+The high `NoAvailableWorkers` rate of 20% suggests that there are many tasks running for durations longer than the `poll_interval`.
+For details on analyzing long task execution durations, see the <<task-manager-theory-long-running-tasks,long running tasks>> theory.
+
+[[task-manager-theory-insufficient-throughput]]
+*Theory*:
+{kib} is polling as frequently as it should, but that isn't often enough to keep up with the workload
+
+*Diagnosis*:
+Evaluating the health stats, you can see the following output of `drift` and `load` under `stats.runtime.value`:
+
+[source,json]
+--------------------------------------------------
+{
+  "drift": { # <1>
+    "p50": 99,
+    "p90": 1245,
+    "p95": 1845,
+    "p99": 2878
+  },
+  "load": { # <2>
+    "p50": 0,
+    "p90": 0,
+    "p95": 10,
+    "p99": 20
+  },
+}
+--------------------------------------------------
+<1> `drift` shows us that at least 95% of tasks are running within 2 seconds of their scheduled time.
+<2> `load` shows us that Task Manager is idle at least 90% of the time, and never uses more than 20% of its available workers.
+
+You can infer from these stats that this {kib} has plenty of capacity, and any delays you might be experiencing are unlikely to be addressed by expanding the throughput.
+
+Suppose the output of `drift` and `load` was the following:
+
+[source,json]
+--------------------------------------------------
+{
+  "drift": { # <1>
+    "p50": 2999,
+    "p90": 3845,
+    "p95": 3845.75,
+    "p99": 4078
+  },
+  "load": { # <2>
+    "p50": 80,
+    "p90": 100,
+    "p95": 100,
+    "p99": 100
+  }
+}
+--------------------------------------------------
+<1> `drift` shows us that all tasks are running 3 to 4 seconds after their scheduled time.
+<2> `load` shows us that at least half of the time Task Manager is running at a load of 80%.
+
+You can infer from these stats that this {kib} is using most of its capacity, but seems to keep up with the work most of the time.
+This assessment is based on the following:
+
+* The `p90` of `load` is at 100%, and `p50` is also quite high at 80%. This means that there is little to no room for maneuvering, and a spike of work might cause Task Manager to exceed its capacity. 
+* Tasks run soon after their scheduled time, which is to be expected. A `poll_interval` of `3000` milliseconds would often experience a consistent drift of somewhere between `0` and `3000` milliseconds. A `p50 drift` of `2999` suggests that there is room for improvement, and you could benefit from a higher throughput.
+
+For details on achieving higher throughput by adjusting your scaling strategy, see <<task-manager-scaling-guidance>>.
+
+[[task-manager-theory-long-running-tasks]]
+*Theory*:
+Tasks run for too long, overrunning their schedule
+
+*Diagnosis*:
+
+The <<task-manager-theory-insufficient-throughput,"Insufficient throughtput to handle the scheduled workload">> theory analyzed a hypothetical scenario where both _drift_ and _load_ were unusually high.
```

Comment:
If we follow the previous format, this should start on the same line as the word "Diagnosis"

Reply:
fixed 👍